File: KakeiPocket/src/main/java/com/kakeipocket/KakeiPocket/entity/MonthlyPlan.java
package com.kakeipocket.KakeiPocket.entity;

import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "monthly_plans")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MonthlyPlan {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "plan_month", nullable = false)
    private Integer month;

    @Column(name = "plan_year", nullable = false)
    private Integer year;

    @Column(name = "income_target", precision = 15, scale = 2)
    private BigDecimal incomeTarget;

    @Column(name = "saving_target", precision = 15, scale = 2)
    private BigDecimal savingTarget;

    @Column(name = "note")
    private String note;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: kakeipocket-fe/src/types/monthlyPlan.ts
export interface MonthlyPlan {
  id: number;
  month: number;
  year: number;
  incomeTarget: number | null;
  savingTarget: number | null;
  note: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateMonthlyPlanRequest {
  month: number;
  year: number;
  incomeTarget: number | null;
  savingTarget: number | null;
  note: string;
}

export interface UpdateMonthlyPlanRequest {
  month?: number;
  year?: number;
  incomeTarget?: number | null;
  savingTarget?: number | null;
  note?: string;
}
